sound/soc/intel/common/soc-acpi-intel-glk-match.c

```diff
@@ -44,7 +44,7 @@ struct snd_soc_acpi_mach snd_soc_acpi_intel_glk_machines[] = {
 	},
 	{
 		.id = "RTL5682",
-		.drv_name = "glk_rt5682_max98357a",
```

Comment:
the drv_name here isn't for information only, it needs to match the platform device id in machine driver glk_rt5682_max98357a:

static const struct platform_device_id glk_board_ids[] = {
	{
		.name = "glk_rt5682_mx98357a",
		.driver_data =
			(kernel_ulong_t)&glk_audio_card_rt5682_m98357a,
	},
	{ }
};

Reply:
Yes, this changes it to match as it doesn't currently. can you test this? We don't have the devices to reproduce this, we never got it from the OEM and you authored the original change.

Reply:
I checked the code history. Main branch shrunk the string but the change did not go to v4.14 so this defect wasn't caught by that time.

Sorry we don't have the board either. The validation is done by odm.